trie.go

```diff
@@ -61,7 +61,10 @@ func (t *Trie) Add(key string, meta interface{}) *Node {
 // with `key`.
 func (t *Trie) Find(key string) (*Node, error) {
 	node := t.nodeAtPath(key)
-	node = node.Children()[nul]
+
+	if node != nil {
```

Comment:
Instead of this, I'd just move line 69 up before the node.Children() assignment, and then change the return to `return node.Children()[nul], true`